Design/AbsorptionCorrections/GenericShapeAbsorptionCorrection.md

```diff
@@ -0,0 +1,249 @@
+
+## Requirements
+
+### Must Haves
+
+1. Able to define a list of available sample holders per instrument
+1. A simple but extensible method for setting the geometry of the container (sample environment), including selection from the above list
+1. A simple but extensible method for setting the geometry of the sample
+1. A method to set the cross-section properties of the sample
+1. A method to set the cross-section properties of the container
+1. Be able to handle elastic/inelastic scattering
+1. Sample should be able to be offset from centre
+1. Separate algorithms to calculate the correction and then apply them, re Paalman & Pings
+
+## Current Structure
+
+There are currently 16 algorithms that relate to different methods for calculating the absorption corrections:
+
+* AbsorptionCorrection
+* AnnularRingAbsorption
+* AnvredCorrection
+* CuboidGaugeVolumeAbsorption
+* CylinderAbsorption
+* CylinderPaalmanPingsCorrection
+* FlatPlateAbsorption
+* FlatPlatePaalmanPingsCorrection
+* HRPDSlabCanAbsorption
+* IndirectAnnulusAbsorption
+* IndirectCylinderAbsorption
+* IndirectFlatPlateAbsorption
+* MayersSampleCorrection
+* MonteCarloAbsorption
+* PearlMCAbsorption
+* SphericalAbsorption
+
+Most are numerical methods defined for some specific shape. The `MonteCarloAbsorption` algorithm in theory has all of the required parts to allow
+the absorption to be computed but it currently only works for elastic data. Also, the can environment must be defined in C++.
+
+## Proposal
+
+The proposed solution is centered on using the Monte Carlo approach to provide the calculation engine for the corrections. This will allow it to
+work for any shape. The complexity involved mostly surrounds the interface required for a user to enter the details regarding the sample & environment.
+
+## Solution Details
+
+The key parts of the proposal are described in the following sections:
+
+* [Running the calculation](#S-running-calculation)
+* [Defining shapes & materials](#S-defining-shapes-and-materials)
+* [Beam characteristics](#S-beam-characteristics)
+
+## <a name="S-running-calculation"></a> Running the calculation
+
+The main calculation will take place in the existing [`MonteCarloAbsorption`](http://docs.mantidproject.org/nightly/algorithms/MonteCarloAbsorption-v1.html), algorithm
+which is currently able to cope with sample/can environments. Upgrades required:
+
+* support for inelastic data
+
+Users will be given a new algorithm called `CalculateSampleCorrection` that will form the interface to running different correction algorithms. These algorithms will assume
+that the metadata required, such as geometry and material properties is all completely defined on the workspace. An example of running the algorithm would be:
+
+```python
+# assume w1 has metadata already set
+w1 = ConvertUnits(w1, Target="Wavelength")
+w1_abs = CalculateSampleCorrection(w1, Method="MonteCarlo",
+    MethodArgs={"NLambda": 500, "NEvents": 300})
```

Comment:
I think that this new property with `dict` interface would be incredibly useful not only for this design but in many places where the code would be nicer and more Pythonic. How would you handle the types of the dictionary entries? Using existing property types?